scripts/run_presubmit_checks_test.py

```diff
@@ -0,0 +1,152 @@
+                run_presubmit_checks.main(args=['--b', specified_branch])
+
+        for script in self.scripts_called:
+            self.assertTrue(script)
+        self.assertIn('Linting passed.', self.print_arr)
+        self.assertIn(
+            'Comparing the current branch with %s' % specified_branch,
+            self.print_arr)
+        self.assertIn('Frontend tests passed.', self.print_arr)
+        self.assertIn('Backend tests passed.', self.print_arr)
+
+    def test_run_presubmit_checks_when_current_branch_exists_on_remote_origin(
+        self) -> None:
+        cmd_to_get_all_changed_file = [
+        'git', 'diff', '--cached', '--name-only', '--diff-filter=ACM',
+        'origin/%s' % self.current_branch]
+        def mock_check_output(cmd: list[str]) -> str:
+            if cmd == self.cmd_to_check_current_branch:
+                return self.current_branch
+            elif cmd == self.cmd_to_match_current_branch_with_remote:
+                return '1'
+            elif cmd == cmd_to_get_all_changed_file:
+                return self.changed_frontend_file
+            else:
+                return ''
+
+        swap_check_output = self.swap(
+            subprocess, 'check_output', mock_check_output)
+        with self.print_swap, swap_check_output, self.swap_pre_commit_linter:
+            with self.swap_backend_tests, self.swap_frontend_tests:
+                run_presubmit_checks.main(args=[])
+
+        for script in self.scripts_called:
+            self.assertTrue(script)
+        self.assertIn('Linting passed.', self.print_arr)
+        self.assertIn(
+            'Comparing the current branch with origin/%s' % self.current_branch,
+            self.print_arr)
+        self.assertIn('Frontend tests passed.', self.print_arr)
+        self.assertIn('Backend tests passed.', self.print_arr)
+
+    def test_frontend_tests_are_not_run_when_no_frontend_files_are_changed(
+        self) -> None:
+        def mock_check_output(cmd: list[str]) -> str:
+            if cmd == self.cmd_to_check_current_branch:
+                return self.current_branch
+            elif cmd == self.cmd_to_match_current_branch_with_remote:
+                return '0'
+            else:
+                return ''
+
+        swap_check_output = self.swap(
+            subprocess, 'check_output', mock_check_output)
+        with self.print_swap, swap_check_output, self.swap_pre_commit_linter:
+            with self.swap_backend_tests, self.swap_frontend_tests:
+                run_presubmit_checks.main(args=[])
+
+        self.assertFalse(self.scripts_called['run_frontend_tests'])
+        self.assertTrue(self.scripts_called['pre_commit_linter'])
+        self.assertTrue(self.scripts_called['run_backend_tests'])
+        self.assertIn('Linting passed.', self.print_arr)
+        self.assertIn(
+            'Comparing the current branch with develop', self.print_arr)
+        self.assertIn('Backend tests passed.', self.print_arr)
```

Comment:
can you also assert that the frontend tests message is *not* present?

Reply:
Done!